### PR TITLE
Improvement to the testbed application

### PR DESCRIPTION
**Pull request by pierre-dejoue**
Fix some warnings, and improve the test data file format to add new optional sections for holes and Steiner points.

See `dude.dat` and `steiner.dat` for examples of the new sections.

Previously the holes in the `dude.dat` example were hardcoded in the application itself! (So, fix that.)

### CODE REVIEW

File: testbed/main.cc [hunk 1]
```diff
@@ -83,18 +89,18 @@ bool random_distribution = false;
...
 
-template <class C> void FreeClear( C & cntr ) {
-    for ( typename C::iterator it = cntr.begin();
-              it != cntr.end(); ++it ) {
-        delete * it;
-    }
-    cntr.clear();
+template <class C>
+void FreeClear( C & cntr ) {
+  for (typename C::iterator it = cntr.begin();  it != cntr.end(); ++it) {
```
Comment (jhasse) [inline testbed/main.cc:94]: Why this change?

File: testbed/main.cc [hunk 2]
```diff
@@ -83,18 +89,18 @@ bool random_distribution = false;
...
+    delete * it;
+  }
+  cntr.clear();
 }
 
 int main(int argc, char* argv[])
 {
-
-  int num_points = 0;
+  string filename = "";
```
Comment (jhasse) [inline testbed/main.cc:102]: No need to initialize std::string

File: testbed/main.cc [hunk 3]
```diff
@@ -207,27 +170,115 @@ int main(int argc, char* argv[])
...
+            break;
+          case Steiner:
+            out_steiner.push_back(new Point(x, y));
+            break;
+          default:
+            assert(0);
+          }
+        }
+      }
+      myfile.close();
```
Comment (jhasse) [inline testbed/main.cc:254]: unneeded as the destructor does this

File: testbed/main.cc [hunk 4]
```diff
@@ -33,19 +33,27 @@
...
 #include <list>
+#include <numeric>
 #include <sstream>
 #include <string>
 #include <vector>
 
 using namespace std;
 using namespace p2t;
 
+using Points = vector<Point*>;
```
Comment (jhasse) [inline testbed/main.cc:52]: Why? This makes the code harder to understand and `Points` isn't that much shorter.

**Review by jhasse — COMMENTED** [review]
Thanks! Why did you change some float functions to double?

File: testbed/main.cc (hunk 1 above)
Comment (pierre-dejoue) [reply testbed/main.cc:94]: Homogenize the style in this file.
- That method used 4 space indentation.
- The line break in the middle of the for() is unwarranted, The line isn't long.
- Line break after template declation

File: testbed/main.cc (hunk 2 above)
Comment (pierre-dejoue) [reply testbed/main.cc:102]: Good catch!

File: testbed/main.cc (hunk 4 above)
Comment (pierre-dejoue) [reply testbed/main.cc:52]: No problem in removing those using statements.
I find it clearer on the contrary to name data structures used in many places, but I understand your point of view as well. Also taking into account the fact that the poly2tri API does not provide those abstractions and we should keep a similar style in the testbed.

File: testbed/main.cc (hunk 3 above)
Comment (pierre-dejoue) [reply testbed/main.cc:254]: True! and we can remove it also at line 232

**Comment by pierre-dejoue** [comment]
Regarding the changes float->double of some of the functions: this is to account for compilation warnings, on my platform at least (I build with MSVC).

New iteration pushed taking into account your comments. Thanks for the review!

**Comment by jhasse** [comment]
Thank you for the PR :)
